README: add slack to README and spruce things up

jess-edwards opened the pull request:
Adds slack channel under the Support section in the read.me

Can someone help me update the logo? I'm not sure where to upload it.

Release justification: Non-production change.
Release note: None

jess-edwards committed:
README: add slack to README, update logo

Release justification: Non-production change.
Release note: None

cockroach-teamcity commented:
[![CLA assistant check](https://cla.crdb.dev/pull/badge/not_signed)](https://cla.crdb.dev/cockroachdb/cockroach?pullRequest=46245) <br/>Thank you for your submission! We really appreciate it. Like many open source projects, we ask that you sign our [Contributor License Agreement](https://cla.crdb.dev/cockroachdb/cockroach?pullRequest=46245) before we can accept your contribution.<br/><sub>You have signed the CLA already but the status is still pending? Let us [recheck](https://cla.crdb.dev/check/cockroachdb/cockroach?pullRequest=46245) it.</sub>

cockroach-teamcity commented:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.png" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cockroachdb/cockroach/46245)
<!-- Reviewable:end -->

irfansharif commented on README.md:
```diff
@@ -74,6 +74,8 @@ CockroachDB supports the PostgreSQL wire protocol, so you can use any available
   Learn how to troubleshoot common errors, cluster and node setup, and SQL query behavior,
   and how to use debug and error logs.
 
+- #[CockroachDB Community Slack](https://cockroa.ch/slack)
```
Remove the "#" and the new line below.

irfansharif reviewed (approved):
LGTM. If by logo you mean the top level cockroachdb one, you're looking for `docs/media/cockroach_db.png`.

jess-edwards commented:
@irfansharif could you help me make these changes and merge the PR? I don't have permissions.

irfansharif committed:
README: spruce up our README page

Include guidance for external contributors, align images, clean up
whitespace, etc.

Release justification: Non-production code changes.
Release note: None.

rafiss reviewed (approved):
> Reviewed 2 of 2 files at r2.
> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/46245)* status: :shipit: complete! 0 of 0 LGTMs obtained (waiting on @jess-edwards, @jordanlewis, and @rafiss)



<!-- Sent from Reviewable.io -->

rafiss reviewed (approved):
lgtm

> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/46245#-:-M3MGMsf5ugDavhE6Lqq:bb7k0qh)* status: :shipit: complete! 0 of 0 LGTMs obtained (waiting on @jess-edwards, @jordanlewis, and @rafiss)



<!-- Sent from Reviewable.io -->

irfansharif commented:
bors r+

irfansharif commented:
bors retry

jess-edwards commented:
oh man. bors says no today.

irfansharif commented:
bors retry
